### PR TITLE
add mo_columns a composite pk

### PR DESCRIPTION
**Pull request by aptend**
## What type of PR is this?

- [ ] API-change
- [ ] BUG
- [x] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

## Which issue(s) this PR fixes:

issue #

## What this PR does / why we need it:

- add mo_columns a composite pk
- remove redundant type define codes in system schema

### CODE REVIEW

File: pkg/vm/engine/tae/catalog/model.go [hunk 1]
```diff
@@ -147,6 +160,16 @@ func init() {
 		Oid:  types.T_timestamp,
 		Size: 8,
 	}
+	tvarchar := types.Type{
+		Oid:   types.T_varchar,
+		Size:  24,
+		Width: 100,
+	}
+	tsinglechar := types.Type{
```
Comment (fengttt) [inline pkg/vm/engine/tae/catalog/model.go:168]: This is not correct.   T_char type is also veriable length.

In fact, I suggest we change all these types to tu32 := types.T_uint32.ToType(),    tsinglechar = types.T_char.ToType(), etc.

**Review by fengttt — COMMENTED** [review]
Types, need to work with some decipline.

File: pkg/vm/engine/tae/catalog/model.go (hunk 1 above)
Comment (aptend) [reply pkg/vm/engine/tae/catalog/model.go:168]: yeah, using ToType() makes it clearer

**Review by XuPeng-SH — APPROVED** [review]
lgtm

**Pull request merged**